### PR TITLE
Add missing frontend tests and enforce code coverage

### PR DESCRIPTION
**Pull request by robertknight**
This PR adds test cases to cover the remaining statements that were not previously covered in the frontend code, bringing coverage up to 100%. With all statements fully covered, enforcing code coverage going forwards becomes straightforward and does not require a service like Codecov. All statements must be either covered or explicitly ignored (with `/* istanbul ignore next */`). If there is code without coverage then `gulp test` will fail, but still output the same information as if the check was not enabled. This means that during development you can just ignore the error.

- Add missing test cases for various modules under the `frontend_apps` directory
- Exclude files in the `test-util` directory from code coverage. These are helpers used only within tests.
- Add `thresholds` property to `coverageIstanbulReporter` in `karma.config.js` to requirement complete statement coverage

### CODE REVIEW

File: lms/static/scripts/postmessage_json_rpc/client.js [hunk 1]
```diff
@@ -36,7 +31,7 @@ async function call(
   params = [],
   timeout = 2000,
   window_ = window,
-  id = generateId()
+  id = generateHexString(10)
```
Comment (robertknight) [inline lms/static/scripts/postmessage_json_rpc/client.js:34]: The `generateId` helper wasn't needed. It's just a trivial function call and the usage is obvious from the context.

File: lms/static/scripts/frontend_apps/components/FilePickerApp.js [hunk 2]
```diff
@@ -45,7 +45,8 @@ function contentDescription(content) {
     case 'vitalsource':
       return 'Book from VitalSource';
     default:
-      return 'Other content';
+      /* istanbul ignore next */
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/components/FilePickerApp.js:48]: I presume this is an in-development/prototype path so doesn't need coverage?

File: lms/static/scripts/frontend_apps/components/test/SubmitGradeForm-test.js [hunk 3]
```diff
@@ -123,21 +123,28 @@ describe('SubmitGradeForm', () => {
...
-          validateGrade: sinon.stub().returns('err'),
-        },
-      });
       const wrapper = renderForm();
       wrapper.find('button[type="submit"]').simulate('click');
       assert.isTrue(wrapper.find('ValidationMessage').prop('open'));
       assert.equal(wrapper.find('ValidationMessage').prop('message'), 'err');
+
+      // Clicking the error message should dismiss it.
+      wrapper.find('ValidationMessage').invoke('onClose')();
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/components/test/SubmitGradeForm-test.js:146]: I've never encountered `wrapper.invoke` before (though it's totally there in Enzyme's docs). Interesting! 

Also, I _feel_ like this is one of those tests that you'd nudge me about using `act` for. Can you help me understand why it's not the proper convention here?

File: lms/static/scripts/frontend_apps/utils/google-picker-client.js [hunk 4]
```diff
@@ -2,7 +2,16 @@ import { loadLibraries } from './google-api-client';
 
 export const GOOGLE_DRIVE_SCOPE = 'https://www.googleapis.com/auth/drive';
 
-/** @param {string} origin */
+/**
+ * Convert a domain (example.com) into a valid web origin (https://example.com).
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/utils/google-picker-client.js:6]: Very helpful, especially given that the function is called `addHttps` and it isn't immediately clear that what it's doing is generating an origin!

**Review by lyzadanger — APPROVED** [review]
Cool, LGTM!

File: lms/static/scripts/frontend_apps/components/FilePickerApp.js (hunk 2 above)
Comment (robertknight) [reply lms/static/scripts/frontend_apps/components/FilePickerApp.js:48]: This code path existed only to convince TypeScript that the function always returns a string, if it returns a value. Throwing an error is a more obvious way to make it clear that the code path should never be executed.

File: lms/static/scripts/frontend_apps/components/test/SubmitGradeForm-test.js (hunk 3 above)
Comment (robertknight) [reply lms/static/scripts/frontend_apps/components/test/SubmitGradeForm-test.js:146]: `wrapper.invoke` is a recent addition to Enzyme. It is effectively shorthand for:

```js
act(() => {
  wrapper.prop(<propName>)(...args);
});
wrapper.update();
```

**Pull request merged**